sympy/functions/elementary/complexes.py

```diff
@@ -468,12 +492,13 @@ def eval(cls, arg):
                 return (-base)**re(exponent)*exp(-S.Pi*im(exponent))
         if isinstance(arg, exp):
             return exp(re(arg.args[0]))
-        if arg.is_zero:  # it may be an Expr that is zero
-            return S.Zero
-        if arg.is_nonnegative:
-            return arg
-        if arg.is_nonpositive:
-            return -arg
+        if arg.is_number or arg.is_Symbol or isinstance(arg, (cls, Unit)):
```

Comment:
Maybe we should use `isinstance(arg, Atom)` instead of `isinstance(arg, Symbol)`. Or use `not arg.args` (see https://github.com/sympy/sympy/wiki/Args-Invariant).